CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+-  A link to any diagnostics (e.g. if using `calicoctl`, you can gathered
+   diagnostics using `calicoctl diags` - this provides instructions for 
+   uploading the diags bundle to transfer.sh - or alternatively if the 
+   diagnostics contains sensitive information we can set up an alternative
+   method for transfer
+-  If using `calicoctl` the output from `calicoctl status` run on each node
+   might also be useful.
+-  Details of your OS
+-  Environment details such as GCE, bare metal, VirtualBox.
+   
+
+## Contributing: Code and Documentation
+
+For contributing code and documentation we follow the GitHub pull request
+model. 
+
+-  Fork the repository on GitHub
+-  Make changes to your local repository in a separate branch
+-  Test the change
+-  Create a pull request which will then go through a review process by one or
+   more of the core team.
+   
+### Testing your changes
+
+If you create a pull request, our automated UT and STs will be run over your
+change.  We will not accept changes that do not consistently pass our automated
+test suites. It is vital that our master branch be passing tests at all times.
+If you tests are failing the automated tests and you don't believe they should
+be, you may need to rebase your branch off the latest master.
+
+Read the [Building and testing calico-containers images guide](docs/Building.md)
+for details on running the UTs and STs.
+
+Where possible, please add any additional tests to ensure we maintain a healthy
+code and feature coverage.
+
+### Documentation
+
+If your code change requires some documentation updates, please include these
+in the pull request.
+
+In particular changes to the `calicoctl` UX will also require changes to the
+`calcoctl` documentation and possibly some of the sample guides.
+
+### Review and merge
+
+Assuming your code review is finished and tests are passing, your change will
+then be merged as soon as possible!  Occasionally we will sit on a change, for
+instance if we are planning to tag a release shortly, but this is only to
+ensure the stability of the branch. After the release we will merge your change
+promptly.
+
+Before merging we prefer that you squash the commits into a single commit to
+ensure we have a cleaner commit history.
+
+### Coding Style
+
+The majority of the code is written in Python and we generally follow the
+[PEP-8 coding style](https://www.python.org/dev/peps/pep-0008).
+
+### Format of the Commit Message
+
+The commit message should include a short subject on what changed, and a body
+detailing why the change was necessary, and any key points about the
+implementation.
+
+Try to keep the subject line no longer than 70 characters, and the lines of the
+body no longer than 80 characters.  This improves readability in both GitHub 
+and when using git tools.
+
+If the pull request fixes an issue, please include details in the body of which
+issue was fixed.  This should be a separate line and of the following format:
```

Comment:
These two sentences kind of contradict.  "includes details" and "This should be a separate line [with 2 words]..."

Maybe just combine them to be:
"If the pull request fixes an issue, include a separate line in the description with the following format:"